### PR TITLE
Parameter store in CMS-utils

### PR DESCRIPTION
**Pull request by parvezalm71**

### CODE REVIEW

File: connector/aws/ssm/ssm.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ssm"
+	"os"
+)
+
+var AwsSsmRegion = os.Getenv("AWS_REGION")
```
Comment (deepikaazadhappay) [inline connector/aws/ssm/ssm.go]: AwsSsmRegion  -->  AWS_SSM_REGION,  keep the variable name in the same format, some are in camel case and some in snake case.

File: connector/aws/ssm/ssm.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+var ssmConfig = &aws.Config{
+	Region:      aws.String(AwsSsmRegion),
+	Credentials: credentials.NewStaticCredentials(AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, "")}
+
+func GetParametersByPath(path string) (params []*ssm.Parameter, err error) {
+	// Error handling
+	if path == EmptyString {
+		err = fmt.Errorf("empty prefix path provided for getting parameters : %s", err)
+		panic(err)
```
Comment (deepikaazadhappay) [inline connector/aws/ssm/ssm.go]: Why panic is required here?

File: connector/aws/ssm/ssm.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+	if err != nil {
+		err = fmt.Errorf("%s | Error in retrieving parameter from parameter store", err)
+		return
+	}
+	return
+}
+func SetEnvironmentVariable() {
+	params, err := GetParametersByPath(SSM_PREFIX)
+	if err != nil {
+		err = fmt.Errorf("failed to get parameter from parameter store. Unable to set environment variables: %s", err)
```
Comment (kavyahathwar) [inline connector/aws/ssm/ssm.go:55]: Do we need to return this error and handle it in the calling function ??

File: connector/aws/ssm/ssm.go [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+		return
+	}
+	for _, param := range params {
+		parameterName := aws.StringValue(param.Name)[len(SSM_PREFIX)+1:]
+		err = os.Setenv(parameterName, *param.Value)
+		if err != nil {
+			return
```
Comment (kavyahathwar) [inline connector/aws/ssm/ssm.go:63]: can we log the error before returning?

File: connector/aws/ssm/ssm.go [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+		return
+	}
+	ssmClient := ssm.New(ssmSession)
+
+	// Get the parameters by path from Parameter Store
+	withDecryption := true
+	err = ssmClient.GetParametersByPathPages(&ssm.GetParametersByPathInput{
+		Path:           &path,
+		Recursive:      aws.Bool(true),
+		WithDecryption: &withDecryption,
```
Comment (kavyahathwar) [inline connector/aws/ssm/ssm.go]: can we not pass true here directly ??

File: connector/aws/ssm/ssm.go (hunk 2 above)
Comment (happay) [reply connector/aws/ssm/ssm.go]: @parvezalm71 Can you fix this?

**Comment by happay** [comment]
This wont be required anymore. The CI/CD pipeline will ensure it fetch all the env variables from AWS SSM and set it as env variable on the docker image.

**Pull request closed without merging**